### PR TITLE
V22F-131 Data anonymous

### PR DESCRIPTION
**Pull request by shoekzema**
added a hash-function

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp [hunk 1]
```diff
@@ -198,13 +198,18 @@ static void SetUserID(void*)
...
     mediator->SetTask(m_task);
     mediator->SetIndicatorSettings(m_indicators);
     mediator->SetInterventionType(m_interventionType);
     mediator->SetMaxTime(m_maxTime);
-    mediator->SetUserID(m_userID);
+
+    // Save the encrypted userID in the SDAConfig
+    char buf[32];
+    size_t userID_encrypted = std::hash<std::string>{}(m_userID);
+    sprintf(buf, "%d", userID_encrypted);
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp:211]: Variable naming must be camelCase and dont capitalize whole abbreviations(ID)
```suggestion
    size_t encryptedUserId= std::hash<std::string>{}(m_userId);
    sprintf(buf, "%d", encryptedUserId);
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp [hunk 2]
```diff
@@ -189,7 +189,7 @@ static void SetMaxTime(void*)
 /// @brief Handle input in the UserID textbox
 static void SetUserID(void*)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp:190]:
```suggestion
static void SetUserId(void*)
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp [hunk 3]
```diff
@@ -189,7 +189,7 @@ static void SetMaxTime(void*)
...
 {
-    char* m_userID = GfuiEditboxGetString(s_scrHandle, m_userIDId);
+    m_userID = GfuiEditboxGetString(s_scrHandle, m_userIDId);
     GfuiEditboxSetString(s_scrHandle, m_userIDId, m_userID);
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp:193]: Don't capitalize abbreviation ID and I would personally give the identifiers referencing a GUI control a clearer name than just 'id'. For example: userIdControl
```suggestion
    m_userId = GfuiEditboxGetString(s_scrHandle, m_userIdControl);
    GfuiEditboxSetString(s_scrHandle, m_userIdControl, m_userId);
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp [hunk 4]
```diff
@@ -198,13 +198,18 @@ static void SetUserID(void*)
...
     mediator->SetIndicatorSettings(m_indicators);
     mediator->SetInterventionType(m_interventionType);
     mediator->SetMaxTime(m_maxTime);
-    mediator->SetUserID(m_userID);
+
+    // Save the encrypted userID in the SDAConfig
+    char buf[32];
+    size_t userID_encrypted = std::hash<std::string>{}(m_userID);
+    sprintf(buf, "%d", userID_encrypted);
+    mediator->SetUserID(buf);
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp:212]: Note that this is a stack allocated variable and thus if user id saves it the reference will be invalid after this function

**Review by suppergerrie2 — APPROVED** [review]

**Review by larsdekwant — APPROVED** [review]

**Pull request merged**